pkg/logql/ast_test.go

```diff
@@ -332,26 +332,70 @@ func TestStringer(t *testing.T) {
 }
 
 func BenchmarkContainsFilter(b *testing.B) {
-	expr, err := ParseLogSelector(`{app="foo"} |= "foo"`, true)
-	if err != nil {
-		b.Fatal(err)
+	lines := [][]byte{
+		[]byte("hello world foo bar"),
+		[]byte("bar hello world for"),
+		[]byte("hello world foobar and the bar and more bar until the end"),
+		[]byte("hello world foobar and the bar and more bar and more than one hundred characters for sure until the end"),
+		[]byte("hello world foobar and the bar and more bar and more than one hundred characters for sure until the end and yes bar"),
 	}
 
-	p, err := expr.Pipeline()
-	if err != nil {
-		b.Fatal(err)
+	benchmarks := []struct {
+		name string
+		expr string
+	}{
+		{
+			"AllMatches",
+			`{app="foo"} |= "foo" |= "hello" |= "world" |= "bar"`,
+		},
+		{
+			"OneMatches",
+			`{app="foo"} |= "foo" |= "not" |= "in" |= "there"`,
+		},
+		{
+			"MixedFiltersTrue",
+			`{app="foo"} |= "foo" != "not" |~ "hello.*bar" != "there" |= "world"`,
+		},
+		{
+			"MixedFiltersFalse",
+			`{app="foo"} |= "baz" != "not" |~ "hello.*bar" != "there" |= "world"`,
+		},
```

Comment:
This case is actually slower than the `main` branch because `|= "baz"` might be tested after `!= "not"`, `!= "there"`.

Reply:
I've found out why. The accumulation was in the wrong order. It was right to left but we need left to right.